### PR TITLE
Add compartments cmt to regimens in nm_to_regimen

### PR DESCRIPTION
**Pull request by dominic-irx**
This PR changes nm_to_regimen so that it uses a map for compartment to dose type. This is useful for mipdeval, where this is needed.

This breaks backward compatibility because we now want an explicit declaration of the type of dosing instead of inferring from presence/absence of columns.

This function is used in two places:

1. PKPDposterior::nonmem_to_stan_data:40, which we haven't touched in a while
2. mipdeval::parse_input_data:45, which is where we have the temporary fix

### CODE REVIEW

File: R/nm_to_regimen.R [hunk 1]
```diff
@@ -19,29 +23,44 @@ nm_to_regimen <- function(data,
...
       tmp$time <- tmp$time - min(tmp$time)
     }
-    if(type == "infusion") {
-      reg[[i]] <- new_regimen(amt = tmp$amt, times = tmp$time, type = "infusion", t_inf = tmp$amt / tmp$rate)
+    if (!is.null(dose_cmts)){
+      # map cmt to dose type
+      reg[[i]] <- new_regimen(
+        amt = tmp$amt,
+        times = tmp$time,
+        cmt = tmp$cmt,
```
Comment (jasmineirx) [inline R/nm_to_regimen.R]: I think we could skip this argument. if it is null, `PKPDsim::sim` will use the compartment definitions of the model file, which is probably safer.

Comment (jasmineirx) [reply R/nm_to_regimen.R]: Could you help me understand this case with an example? It is possible that NONMEM-style data and PKPDsim model definitions use different compartments for one dose type.

Comment (dominic-irx) [reply R/nm_to_regimen.R]: https://github.com/InsightRX/mipdeval/pull/25/files

This is the **only** relevant place this function is currently being used, where we want to specify which compartments are from which doses using the NONMEM data set. Leaving it to be assumed by the model downstream, especially when CMT should be specified in the data set anyway, is not optimal imo.

File: R/nm_to_regimen.R [hunk 2]
```diff
@@ -19,29 +21,57 @@ nm_to_regimen <- function(data,
...
+        # if no RATE is given, then assume bolus
+        suppressWarnings({
+          reg[[i]] <- new_regimen(
+            amt = tmp$amt,
+            times = tmp$time,
+            cmt = tmp$cmt
+          )
+        })
+      }
+    } else if ("rate" %in% colnames(doses) &! 0 %in% doses$rate){
```
Comment (roninsightrx) [inline R/nm_to_regimen.R]: What if we have two doses: on a first dose RATE=0 (bolus loading) and on second dose RATE=1? Then all doses will be treated as bolus, I think. Would add that as a test case to check, and potentially update behavior?

**Review by roninsightrx — COMMENTED** [review]
looks good, but one last question about edge case

**Review by roninsightrx — APPROVED** [review]
lgtm
